templates/_helpers.tpl

```
@@ -60,10 +60,10 @@ Usage:
 
 {{- define "render.secret-name" -}}
   {{- if .value -}}
-    {{- if kindIs "string" .value -}}
-{{ .defaultName }}
-    {{- else -}}
-{{ .value.valueFrom.secretKeyRef.name }}   
+    {{- if kindIs "map" .value -}}
+{{ .value.valueFrom.secretKeyRef.name }}
+    {{- else if kindIs "string" .value -}}
+{{ .defaultName }}  
```

Add else where you can throw an error that provided value is not correct.